setup/lib/utils.js

```diff
@@ -0,0 +1,74 @@
+const { setFailed } = require('@actions/core');
+const os = require('os');
+const path = require('path');
+const execSync = require('child_process').execSync;
+const semver = require('semver');
+const core = require('@actions/core');
+
+// arch in [arm, x32, x64...] (https://nodejs.org/api/os.html#os_os_arch)
+// return value in [amd64, 386, arm]
+function mapArch(arch) {
+    const mappings = {
+        x32: '386',
+        x64: 'amd64'
+    };
+    return mappings[arch] || arch;
+}
+
+// os in [darwin, linux, win32...] (https://nodejs.org/api/os.html#os_os_platform)
+// return value in [darwin, linux, windows]
+function mapOS(os) {
+    const mappings = {
+        darwin: 'macOS',
+        win32: 'windows'
+    };
+    return mappings[os] || os;
```

Comment:
We may need an exception for unknown `os`

Reply:
Are you thinking of something like this?

```
function mapOS(os) {
    const mappings = {
        darwin: 'macOS',
        win32: 'windows'
    };

    if (os in mappings) {
        return mappings[os];
    } else {
        return 'Unknown OS';
    }
}
```

Reply:
This might also be a good candidate for the refactor work once the PR is merged.

Reply:
Yes, similar to your example code. We also need to consider the case for `linux`

```typescript
function getPlatform(): string {
    const platform: string = os.platform();
    switch (platform) {
        case 'linux':
            return 'linux';
        case 'darwin':
            return 'darwin';
        case 'win32':
            return 'windows';
        default:
            throw new Error(`Unsupported platform: ${platform}`);
    }
}
```

Reply:
Issue created: https://github.com/notaryproject/notation-action/issues/13